tools/rosconsole/src/rosconsole/rosconsole.cpp

```diff
@@ -425,6 +435,25 @@ void initialize()
     backend::function_notifyLoggerLevelsChanged = notifyLoggerLevelsChanged;
     backend::function_print = _print;
 
+    char* line_buffered = NULL;
+#ifdef _MSC_VER
+    _dupenv_s(&line_buffered, NULL, "ROSCONSOLE_STDOUT_LINE_BUFFERED");
+#else
+    line_buffered =  getenv("ROSCONSOLE_STDOUT_LINE_BUFFERED");
+#endif
+    if (line_buffered)
+    {
+      try
+      {
+        g_force_stdout_line_buffered = boost::lexical_cast<bool>(std::string(line_buffered));
```

Comment:
I removed lexical cast. In case the value does not equal 1, the default value 0 will be used. If the value provided is neither 0 nor 1, a warning is issued to notify user that something is wrong and he/she needs to check correctness of parameters.